openomni/packet.py

```diff
@@ -66,6 +66,10 @@ def __init__(self, data=""):
             segment_len = min(Packet.MAX_BODY_SEGMENT_LEN,self.body_len,len(data)-14)
             self.body = data[13:(13+segment_len)]
             self.crc = ord(data[13+segment_len])
+            if (self.message_type[0].encode('hex') == '1d'):
+                self.minutes_active = ((ord(data[18]) & 0x0f) << 6) + (ord(data[19]) >> 2)
+                self.reservoir_level = round((float((((ord(data[19]) & 0x03) << 6) + (ord(data[20]) >> 2)))*50)/255, 0)
```

Comment:
I think this belongs in a class specific to status responses.  Packets (and Messages) are container classes for data.

Reply:
My view is that the code should probably be changed to reflect the recent understanding of packets and messages. The packet class should not know/care about message type, byte9, address2, etc - these are part of the message. The 31 bytes of data from each packet should be concatenated into a message. A message consists of 6 header bytes (pod address2, byte9, len), body data, and the CRC-16. The message class should not know about mtype - that's part of the body. A body class should do the actual parsing of the body data, probably subclassed for each mtype.

So there are really three levels of protocol: the packet level (sending 31 byte of data), the message level (sending a block of data with some headers and checksum), and the body level (which is the data as the "application" sees it).

(These comments aren't for this PR specifically, but comments in response to the thread.)